cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/CruiseControlMetricsProcessor.java

```diff
@@ -72,6 +72,10 @@ private void updateCachedNumCoresByBroker(Cluster cluster) {
       // Compute cached number of cores by broker id if they have not been cached already.
       _cachedNumCoresByBroker.computeIfAbsent(brokerId, bid -> {
         Node node = cluster.nodeById(bid);
+        if (node == null) {
+          LOG.error("Received metrics from unrecognized broker {}.", bid);
```

Comment:
Let's make this `warn` -- i.e. it doesn't stop CC from running, but still unexpected.